---
title: C28125
ms.date: 11/04/2016
ms.topic: reference
f1_keywords:
  - "C28125"
helpviewer_keywords:
  - "C28125"
ms.assetid: 3f7b8db4-6a64-4480-919e-3f3ceca7f948
author: mikeblome
ms.author: mblome
manager: markl
ms.workload:
  - "multiple"
---
# C28125
warning C28125: The function must be called from within a try/except block

 The driver is calling a function that must be called from within a try/except block, such as [ProbeForRead](https://docs.microsoft.com/windows-hardware/drivers/ddi/content/wdm/nf-wdm-probeforread), [ProbeForWrite](https://docs.microsoft.com/windows-hardware/drivers/ddi/content/wdm/nf-wdm-probeforwrite), [MmProbeAndLockPages](https://docs.microsoft.com/windows-hardware/drivers/ddi/content/wdm/nf-wdm-mmprobeandlockpages).

## Example
 The following code example generates this warning:

```
ProbeForRead(addr, len, 4);
```

 The following code example avoids this warning:

```

__try
{
    ProbeForRead(addr, len, 4);
}
__except(EXCEPTION_EXECUTE_HANDLER)
{
    Status = GetExceptionCode();
    ... report error status
}
```